sql/core/src/test/scala/org/apache/spark/sql/execution/joins/SemiJoinSuite.scala

```diff
@@ -108,9 +109,26 @@ class SemiJoinSuite extends SparkPlanTest with SharedSQLContext {
     left,
     right,
     condition,
+    LeftSemi,
     Seq(
       (2, 1.0),
       (2, 1.0)
     )
   )
+
+  testLeftSemiJoin(
+    "anti join test",
+    left,
+    right,
+    condition,
+    LeftAnti,
+    Seq(
+      (1, 2.0),
+      (1, 2.0),
+      (3, 3.0),
+      (null, null),
```

Comment:
This is actually what I am struggling, if the join key is `null`, then will we let's go into the anti join result?